Sources/TuistKit/Services/ShareService.swift

```diff
@@ -226,6 +234,57 @@ struct ShareService {
         }
     }
 
+    private func shareIPA(
+        _ appPaths: [AbsolutePath],
+        fullHandle: String,
+        serverURL: URL
+    ) async throws {
+        guard appPaths.count == 1,
+              let ipaPath = appPaths.first else { throw ShareServiceError.multipleAppsSpecified(appPaths.map(\.pathString)) }
+
+        guard let appBundlePath = try fileArchiverFactory.makeFileUnarchiver(for: ipaPath).unzip().glob("**/*.app").first
+        else { throw ShareServiceError.appBundleInIPANotFound(ipaPath) }
+        let appBundle = try await appBundleLoader.load(appBundlePath)
+        let displayName = appBundle.infoPlist.name
+
+        let preview = try await previewsUploadService.uploadPreviews(
```

Comment:
I'd add a log right before this to signal that an asynchronous process is going to start and they might need to wait for the upload to complete. Once we have a component for this, we can show a progress indicator.